### PR TITLE
Explict configuration for deb package build

### PR DESCRIPTION
**Pull request by generall**
- **explicit configuration of deb package**
- **use special build feature**

Bonus: now works with web-ui

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -38,6 +39,7 @@ stacktrace = ["rstack-self"]
 chaos-testing = []
 data-consistency-check = ["collection/data-consistency-check"]
 gpu = ["gpu/gpu", "segment/gpu"]
+deb = []
```
Comment (timvisee) [inline Cargo.toml:42]: Don't we need at least `default` in here?

File: Cargo.toml [hunk 2]
```diff
@@ -282,4 +284,21 @@ opt-level = 3
...
+section = "database"
+priority = "optional"
+extended-description = """\
+Qdrant is a high-performance vector search engine. It is designed to index and search large collections of high-dimensional vectors.\
+"""
+assets = [
+    ["target/release/qdrant", "usr/bin/", "755"],
+    ["README.md", "usr/share/doc/qdrant/README", "644"],
+    ["static/*/*", "/etc/qdrant/static", "644"],
+    ["config/deb.yaml", "etc/qdrant/config.yaml", "644"],
```
Comment (timvisee) [inline Cargo.toml:303]: Only one of the lines has the `/` prefix. Is that intended?

File: Cargo.toml [hunk 3]
```diff
@@ -282,4 +284,21 @@ opt-level = 3
 
 [patch.crates-io]
 # Temporary patch until our PRs are merged.
-tar = { git = "https://github.com/qdrant/tar-rs", branch="main" }
+tar = { git = "https://github.com/qdrant/tar-rs", branch = "main" }
+
+[package.metadata.deb]
+maintainer = "Qdrant Team <team@qdrant.com>"
+depends = "$auto"
+features = ["deb"]
```
Comment (timvisee) [inline Cargo.toml:292]: This is the default, but it makes it a bit more explicit for me that that default features are used as well.

```suggestion
default-features = true
features = ["deb"]
```

File: Cargo.toml (hunk 2 above)
Comment (generall) [reply Cargo.toml:303]: All examples are actually using path without `/`, so I think it is safe to remove

Comment (timvisee) [reply Cargo.toml:303]: It should be a relative path to properly resolve it when installing the package.

```suggestion
    ["target/release/qdrant", "usr/bin/", "755"],
    ["README.md", "usr/share/doc/qdrant/README", "644"],
    ["static/*/*", "etc/qdrant/static", "644"],
    ["config/deb.yaml", "etc/qdrant/config.yaml", "644"],
```

**Review by timvisee — APPROVED** [review]
Builds fine locally. Thanks!

**Pull request merged**